### PR TITLE
add dump()/load() to Parameters class

### PR DESCRIPTION
**Pull request by newville**
This PR gives the Parameters class methods dump/dumps/load/loads to use a JSON representation of the Parameters.  

A few other minor fixes to Parameter class are included, for example that having 'min==max' now raises a ValueError.

### CODE REVIEW

File: lmfit/parameter.py [hunk 1]
```diff
@@ -238,19 +318,19 @@ def setup_bounds(self):
...
+            self.from_internal = lambda val: self.min - 1.0 + sqrt(val*val + 1)
+            _val  = sqrt((self._val - self.min + 1.0)**2 - 1)
         elif self.min in (None, -inf):
             self.from_internal = lambda val: self.max + 1 - sqrt(val*val + 1)
-            _val  = sqrt((self.max - self._val + 1)**2 - 1)
+            _val  = sqrt((self.max - self._val + 1.0)**2 - 1)
         else:
             self.from_internal = lambda val: self.min + (sin(val) + 1) * \
-                                 (self.max - self.min) / 2
+                                 (self.max - self.min) / 2.0
```
Comment (danielballan) [inline lmfit/parameter.py]: We should have `from __future__ import division` to handle this kind of thing, part of the standard header for packages supporting 2 and 3.

**Comment by danielballan** [comment]
This is great.

**Comment by tritemio** [comment]
This is very useful! Thanks.

The only remark I have is that I would rename `dumps` and `loads` to `dumpstr`  `loadstr` (or something similar) to be more explicit.

**Comment by newville** [comment]
@danielballan I added `from __future__ import division` to parameters.py

@tritemio  `dumps()` is what the Pickle and json modules use for "dump to string".  I agree that `dumpstr` is a better name, but am willing to follow the precedence of the stdlib.

**Pull request merged**